homeassistant/components/cover/__init__.py

```diff
@@ -35,10 +35,26 @@
 
 ENTITY_ID_FORMAT = DOMAIN + '.{}'
 
+DEVICE_CLASS_AWNING = 'awning'
```

Comment:
I think we should make a separate PR for this where we also document or link to the updated dev docs.

In particular we should describe what these things are. I don't know the difference of some of these. What's the difference between an awning and a shade?

Reply:
I confirmed with @balloob before opening this PR that it would be fine to combine.  These classes are described in the arch issue and updated in the cover docs.  The [cover dev docs](https://developers.home-assistant.io/docs/en/entity_cover.html) is empty, and I'm happy to flush that out in it's own PR--but let's not let that hold up this change.

Reply:
I don't think it's clear in the docs what these classes are. We have to make clear what type of device each class represents in home assistant. It's not enough that a vendor uses a particular name. We have to define the names we use so that we can categorize correctly independent of vendor names. Dev docs PR should be linked before merging the source PR.

I would prefer not to mix an important change like this with a platform addition. It should be easy to go back and see what changed and reference that PR.

Reply:
Your contribution is welcome to the [arch issue where these classes were identified](https://github.com/home-assistant/architecture/issues/153) if you think they need to be clarified/improved.

Requiring dev docs and separate PRs for core changes are not current requirements to contribute.  Those are great ideas, but if we are going to require that, should be clarified in the check lists so we're all on the same page.  It's frustrating to be given advise by the founder of the project to then have the goal posts moved by someone else.

Reply:
You asked for my review. I'm telling you what I think. I'm sorry that that is frustrating. It's not my intention to be frustrating.

I'll make post in the architecture issue, that I don't know what a shade is and that we can't define it as "anything".
Yes, we should be clearer about requirements. It's not a new idea to require documentation for source contributions. We have an explicit requirement for user docs PR. I don't think we should take less care regarding dev docs.

Reply:
I was hoping you'd focus on the code, less on the process. =)

Dev docs PR created (home-assistant/developers.home-assistant#187) and I removed the describes from the code so we're not doing duel maintenance.